src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -359,28 +427,38 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                                       const ControlKeyStates modifiers,
                                       const bool keyDown)
     {
-        // When there is a selection active, escape should clear it and NOT flow through
-        // to the terminal. With any other keypress, it should clear the selection AND
-        // flow through to the terminal.
+        // Update the selection, if it's present
         // GH#6423 - don't dismiss selection if the key that was pressed was a
         // modifier key. We'll wait for a real keystroke to dismiss the
         // GH #7395 - don't dismiss selection when taking PrintScreen
         // selection.
-        // GH#8522, GH#3758 - Only dismiss the selection on key _down_. If we
-        // dismiss on key up, then there's chance that we'll immediately dismiss
+        // GH#8522, GH#3758 - Only modify the selection on key _down_. If we
+        // modify on key up, then there's chance that we'll immediately dismiss

```

Comment:
\<showerthought>


If we could handle this modification on key up, then could we do something like bind `markMode` to `shift+left`, and have the `shift+left` keydown _enter mark mode_, and the keyup _extend the selection to the left_? Essentially getting quickedit-like selection?

This is of course, crazy, but something to think about

Reply:
HECK IT DOESN"T EVEN MATTER

a user could just bind markmost to shift+left, and that would start a selection at the cursor, and the subsequent keypresses would extend it without hitting the keybinding, which is actually just like conhost. Brilliant 🥂